cmd/swctl/app/cmd_dependency.go

```diff
@@ -312,9 +307,14 @@ func ResizeHugePages(cli Cli, size uint) error {
 	if err != nil {
 		return err
 	}
-	allocatedHP, _ := AllocatedHugePages(cli)
+	allocatedHP, err := AllocatedHugePages(cli)
+
 	if size != uint(allocatedHP) {
-		return errors.New("failed to allocate enough continuous memory")
+		return errors.New(fmt.Sprintf("failed to allocate enough hugepages (%s),successfully allocated %s hugepages, totally continuous memory %d MB ",
+			size, allocatedHP, (allocatedHP*hugePageSize)/100))
+	}
+	if err != nil {
```

Comment:
First you check for error and only if no error is present (err != nil) then you use the returned value (allocatedHP). By default in go the returned value should be handled as undefined if there is an error.